pdata/internal/log_record_flags_test.go

```diff
@@ -0,0 +1,40 @@
+package internal
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLogRecordFlags(t *testing.T) {
+	flags := generateTestLogRecordFlags()
+	assert.True(t, flags.IsSampled())
+
+	flags.SetIsSampled(false)
+	assert.False(t, flags.IsSampled())
+
+	flags.SetIsSampled(true)
+	flags.SetIsSampled(true)
+	assert.True(t, flags.IsSampled())
```

Comment:
Should we add a scenario that calls `flags.SetIsSampled(false)`  twice in a row?

Reply:
Done.